executor/simple.go

```diff
@@ -1296,13 +1296,30 @@ func (e *SimpleExec) executeDropUser(ctx context.Context, s *ast.DropUserStmt) e
 			break
 		}
 
+		// delete from activeRoles
+		if s.IsDropRole {
+			for i := 0; i < len(activeRoles); i++ {
+				if activeRoles[i].Username == user.Username && activeRoles[i].Hostname == user.Hostname {
+					activeRoles = append(activeRoles[:i], activeRoles[i+1:]...)
```

Comment:
Is the `activeRoles` information propagated to other TiDB instance?
If not, when you drop the role in one TiDB instance and query in another TiDB instance, the error remains there.

Reply:
It is a problem, but I don't know how to solve it... I read through `(*SimpleExec).setRoleRegular` and can't find related logic.🤔

Reply:
./executor/simple.go:   return domain.GetDomain(e.ctx).NotifyUpdatePrivilege()


I'm not sure whether this works. E.g, update the privilege cache to make every TiDB instance reload the active role info when drop user is called.

Reply:
I found it hard to do so. In `Domain` I can't find a way to get the related session, thus I don't know how to get the `activeRoles`. If we can get it, I think we can check and apply `activeRoles` in `(*Handle).Update()`.